### PR TITLE
fix(mongoose,typegoose,#881): Allow string objectId filters

### PR DESCRIPTION
**Pull request by doug-martin**

### CODE REVIEW

File: packages/query-typegoose/__tests__/query/comparison.builder.spec.ts [hunk 1]
```diff
@@ -18,6 +20,14 @@ describe('ComparisonBuilder', (): void => {
         },
       });
     });
+
+    it('should convert convert query fields to objectIds if the field is an objectId', (): void => {
```
Comment (smolinari) [inline packages/query-typegoose/__tests__/query/comparison.builder.spec.ts:24]: Small typo. One convert too many. :)

Scott

File: packages/query-mongoose/__tests__/query/comparison.builder.spec.ts [hunk 2]
```diff
@@ -18,6 +19,14 @@ describe('ComparisonBuilder', (): void => {
         },
       });
     });
+
+    it('should convert convert query fields to objectIds if the field is an objectId', (): void => {
```
Comment (smolinari) [inline packages/query-mongoose/__tests__/query/comparison.builder.spec.ts:23]: Small typo. One convert too many. :)

Scott

File: packages/query-mongoose/__tests__/query/comparison.builder.spec.ts [hunk 3]
```diff
@@ -153,6 +162,14 @@ describe('ComparisonBuilder', (): void => {
...
```
Comment (smolinari) [inline packages/query-mongoose/__tests__/query/comparison.builder.spec.ts:166]: Same here. 

Scott

File: packages/query-typegoose/__tests__/query/comparison.builder.spec.ts [hunk 4]
```diff
@@ -153,6 +163,14 @@ describe('ComparisonBuilder', (): void => {
...
```
Comment (smolinari) [inline packages/query-typegoose/__tests__/query/comparison.builder.spec.ts:167]: Here too. Seems a copy and paste mistake. :)

Scott

File: packages/query-typegoose/__tests__/query/comparison.builder.spec.ts [hunk 5]
```diff
@@ -164,6 +182,14 @@ describe('ComparisonBuilder', (): void => {
...
```
Comment (smolinari) [inline packages/query-typegoose/__tests__/query/comparison.builder.spec.ts:186]: And here.

Scott

File: packages/query-mongoose/__tests__/query/comparison.builder.spec.ts [hunk 6]
```diff
@@ -164,6 +181,14 @@ describe('ComparisonBuilder', (): void => {
...
```
Comment (smolinari) [inline packages/query-mongoose/__tests__/query/comparison.builder.spec.ts:185]: Here too. I'll stop with this one. (been going up and down between the two packages as I found it first in the typegoose package). The extra convert is in all the tests.  

Scott

**Review by smolinari — APPROVED** [review]
From a technical standpoint, not much I can add. Just caught the copy and past error with "convert". :)

Scott

**Comment by doug-martin** [comment]
Thank you for reviewing @smolinari and good catch on the test naming. I can't tell you how many times I've looked at a spec name and wondered how that got through :)

**Pull request merged**